### PR TITLE
Propose of logs stream destination and library recomendation

### PR DESCRIPTION
**Pull request by dbadura**
**Description**

Changes proposed in this pull request:

- recommend log stream destination
- recommend logging library  

**Related issue(s)**

### CODE REVIEW

File: internal/proposals/logs/streams/apex.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+	"os"
+)
+
+func main() {
+	log.SetHandler(json.New(os.Stderr))
+	testApexLogger(log.WithField("request_id", "dajdhaskj"))
+}
+
+func testApexLogger(logEntry *log.Entry) {
+	logEntry.Infof("just normal log with msg: %s", "Hello From Zap")
```
Comment (franpog859) [inline internal/proposals/logs/streams/apex.go:15]: I'd rather say "Hello from Apex" ;D

**Review by franpog859 — APPROVED** [review]
Looks cool! 
Shouldn't we at least suggest which of these libraries we'd like to use in the components? I mean just one library so it'd be just a no-brainer for the implementing team. It'd be good to specify all the possibilities for the logging levels

File: internal/proposals/logs/streams/apex.go (hunk 1 above)
Comment (dbadura) [reply internal/proposals/logs/streams/apex.go:15]: Good catch

File: internal/proposals/logs/streams/README.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+# Summary
+
+I would recommend logging everything to stderr, because:
+- zap, zerolog and apex/log and I think other libraries can log everything to stderr 
+- it's possible to filter logs by level and it's not needed to filter the logs by stream
+- requires less work on current components
+
+For new components, the recommendation is `Zap`.
+For already components, we should use what we currently have and unified log messages.
```
Comment (janmedrek) [inline internal/proposals/logs/streams/README.md:87]: I would get rid of that sentence and the next one as well. As far as I know, the vast majority of libraries currently used in Kyma do not support log levels/formats/streams. We want to have the logging unified and using the common library is only making that easier.

File: internal/proposals/logs/streams/README.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+## apex/log
+pros:
+- ability to chain loggers (add context) using `entry` struct
+- library can log in JSON or text format. It has the ability to be extended.
+- configurable level of filtering
+- available log levels: ERROR, INFO, FATAL, DEBUG, WARN
+
+cons:
+- cannot set timestamp format. The default time format looks like this:  `2020-12-22T12:52:39.906885+01:00`
```
Comment (janmedrek) [inline internal/proposals/logs/streams/README.md:68]: Doesn't it eliminate this library from the pool? That's one of our basic assumptions. If it does then please include this information here.

**Review by janmedrek — CHANGES_REQUESTED** [review]

File: internal/proposals/logs/streams/README.md (hunk 3 above)
Comment (dbadura) [reply internal/proposals/logs/streams/README.md:68]: I will add test which proofs that this timestamp is valid with RFC3339 and also add explanation.

File: internal/proposals/logs/streams/README.md (hunk 2 above)
Comment (dbadura) [reply internal/proposals/logs/streams/README.md:87]: I added information that if library can't meet requirements should be exchanged.

**Review by janmedrek — DISMISSED** [review]

**Review by kazydek — APPROVED** [review]
🎉

**Pull request merged**